src/components/rocm/roc_profiler.c

```diff
@@ -805,22 +812,123 @@ evt_id_to_info(uint64_t event_id, event_info_t *info)
     return PAPI_OK;
 }
 
+/** @class rocm_verify_no_repeated_qualifiers
+  * @brief Verify that a user has not added multiple device or instance qualifiers
+  *        to an event name.
+  *
+  * @param *eventName
+  *   User provided event name we need to verify.
+*/
 int
-evt_name_to_device(const char *name, int *device)
+rocm_verify_no_repeated_qualifiers(const char *eventName)
 {
-    char *p = strstr(name, ":device=");
-    if (!p) {
+    int numDeviceQualifiers = 0, numStatsQualifiers = 0;
+    char tmpEventName[PAPI_2MAX_STR_LEN];
+    int strLen = snprintf(tmpEventName, PAPI_2MAX_STR_LEN, "%s", eventName);
+    if (strLen < 0 || strLen >= PAPI_2MAX_STR_LEN) {
+        SUBDBG("Failed to fully write eventName into tmpEventName.\n");
+        return PAPI_EBUF;
+    }
+    char *token = strtok(tmpEventName, ":");
+    while(token != NULL) {
+        if (strncmp(token, "device", 6) == 0) {
+            numDeviceQualifiers++;
+        }
+        else if (strncmp(token, "stat", 4) == 0){
```

Comment:
`instance`

Reply:
Thank you for catching this, I will create a PR for this tomorrow.